### ISSUE TITLE
execute function for Compeltable

### ISSUE THREAD

**Issue by TheLester** [airbnb/MvRx#153]
In `BaseMvRxViewModel.kt` there is `exectute` extension functions only for `Observable` and `Single`. Any plans to support `Completable` as well?

**Comment by hellohuanlin** [comment]
`Async`'s success case comes with a payload. We can use `Unit` for `Completable` if we wanna support it. 
What do you think? @gpeal @BenSchwab @elihart

**Comment by gpeal** [comment]
@hellohuanlin Yeah, using Unit for this makes sense. It would bloat the api though.
As a consumer, you could call `completable.toObservable().execute { ... }` too or create the extension yourself if it is a common case.

I'm 50/50 on this one.

**Comment by alexbchr** [comment]
I already added an execute function in my `BaseViewModel` class to work with Completable using an `Async<Unit>` object. Here is the code based on `Single<T>.execute()` if you're interested: 
```kotlin
protected fun Completable.execute(stateReducer: State.(Async<Unit>) -> State): Disposable {
    setState { this.stateReducer(Loading()) }

    return toSingle { Success(Unit) as Async<Unit> }
        .onErrorReturn { Fail(it) }
        .subscribe { async -> setState { this.stateReducer(async) } }
        .disposeOnClear()
}
```

**Comment by gpeal** [comment]
@BenSchwab @hellohuanlin what do you think here?

**Comment by alexbchr** [comment]
> @alexbchr can you try not to duplicate the logic and directly call the other execute functions, something like
> 
> ```
> fun Completable.execute(reducer: S.(Async<Unit>) -> S) 
>     = toSingle { Unit }.execute(reducer)
> ```
> ^^ semipseudo code

Well your semi-pseudo code is not so pseudo and works well for me! And it is much cleaner to reuse the existing implementation.

**Comment by BenSchwab** [comment]
```
fun Completable.execute(reducer: S.(Async<Unit>) -> S) 
    = toSingle { Unit }.execute(reducer)
```

I would be in favor of shipping this extension with MvRx.
